source/helpers/Misc.ts

```diff
@@ -7,3 +9,56 @@ export function wait(ms: number): Promise<void> {
 export function getScreenHeightProportion(proportion: number): number {
   return Math.floor((Dimensions.get("window").height * proportion) / 100);
 }
+
+export function getAPIEnvironmentIdentifierFromUrl(
+  apiUrl: string
+): string | null {
+  const apiUrlPrefixEnvironmentMapping: Record<string, string> = {
+    dev: "d",
+    release: "r",
+    staging: "s",
+    test: "t",
+  };
+
+  const prefixRegex = /^https:\/\/(?:(.*)\.)?api\.helsingborg.io/i;
+
+  const apiUrlLowercase = apiUrl.toLowerCase();
+  const match = apiUrlLowercase.match(prefixRegex);
+
+  if (match) {
+    const apiPrefix = match[1];
+
+    if (apiPrefix === undefined) {
+      return "p";
+    }
+
+    const acceptablePrefixLetter = apiUrlPrefixEnvironmentMapping[apiPrefix];
+
+    if (typeof acceptablePrefixLetter === "string") {
```

Comment:
Do you really need to check if the type is a string?
Wouldn't this be enough:

```js
if (acceptablePrefixLetter !== undefined) { ... }

or just
if (acceptablePrefixLetter) { ... }
```

Reply:
No, because `apiUrlPrefixEnvironmentMapping` is an `Object`, which means it has internal properties such as `hasOwnProperty`. Not sure if there's an accessor similar to `[name]` that ignores those, but otherwise this is the "workaround" I found that passes the tests.